lib/fs/basicfs_windows.go

```diff
@@ -150,3 +151,31 @@ func (f *BasicFilesystem) Roots() ([]string, error) {
 
 	return drives, nil
 }
+
+func (f *BasicFilesystem) resolveWin83(absPath string) string {
+	if !strings.Contains(absPath, "~") {
+		return absPath
+	}
+	if in, err := syscall.UTF16FromString(absPath); err != nil {
+		out := make([]uint16, 2*len(absPath))
+		n := uint32(len(out))
+		for {
+			n, err = syscall.GetLongPathName(&in[0], &out[0], n)
+			if err != nil {
+				break
+			}
+			if n <= uint32(len(out)) {
+				return syscall.UTF16ToString(out[:n])
+			}
+			out = make([]uint16, n)
```

Comment:
The returned `n` is:

> The size of the buffer lpszLongPath points to, in TCHARs.

Apparently `TCHAR` can mean `char` or `wchar_t` depending on whether your project is compiled as unicode or not. I'm not sure what Go does here, but it's possible that it might actually want a buffer with `2*n` *bytes*. 

If it does, this combined with the loop will throw this function into an infinite loop. I think one of those test things might be a good idea here...